### PR TITLE
PR for loader replacing DOM elements issue. Closes #75

### LINKED ISSUE #75: Loading hides the tabs section
When new images loading loading spinner is removing tabs from the DOM. 
Tabs should be visible to the user when loading images.

image

### PR DESCRIPTION
**Pull request by ashwanijha1996**
## Changes proposed

1. CSS of loader changed to add margin bottom in place of margin top
2. Logic of conditional rendering changed to show gallery/favourites tabs in DOM along with loader.
3. Tab switching logic changed as now i am passing click handler function and state variable to galleryTabs component.
4. start script in package.json changed as it was giving an error - error:0308010C:digital envelope routines::unsupported.

## Check List (Check all the applicable boxes) 

- [ x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x ] This PR does not contain plagiarized content.
- [x ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -51,4 +51,4 @@
     "eslint-plugin-jsx-a11y": "^6.6.1",
     "eslint-plugin-react": "^7.31.11"
   }
-}
+}
```
Comment (furkanmutlu) [inline frontend/package.json:54]: please add empty line at the end

File: frontend/package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "sass": "^1.49.9"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "react-scripts --openssl-legacy-provider start",
```
Comment (furkanmutlu) [inline frontend/package.json:27]: hmm , nice fix, I think I will create a issue to switch from CRA to Vite

File: frontend/src/components/Loading/_Loading.scss [hunk 3]
```diff
@@ -34,11 +34,21 @@
...
-    100% { transform: rotate(360deg); }
-}
+    0% {
+        transform: rotate(0deg);
+    }
+
+    100% {
+        transform: rotate(360deg);
+    }
+}
```
Comment (furkanmutlu) [inline frontend/src/components/Loading/_Loading.scss:54]: Please add empty line at the end, I think you can configure your IDE to do it automatically for you.

**Review by furkanmutlu — CHANGES_REQUESTED** [review]
I have few minor comments, overall your changes looks good and fixing the problems :) thank you for your contribution!

File: frontend/package.json (hunk 1 above)
Comment (ashwanijha1996) [reply frontend/package.json:54]: Hi @furkanmutlu  I tried to add empty line at the end but my IDE is ignoring that. The above mentioned change is due to code formatter i think and it is looking fine to me. But if you still want me to change that then i will change my IDE settings and do that?

File: frontend/src/components/Loading/_Loading.scss (hunk 3 above)
Comment (ashwanijha1996) [reply frontend/src/components/Loading/_Loading.scss:54]: Same as i commented above.

File: frontend/package.json (hunk 1 above)
Comment (furkanmutlu) [reply frontend/package.json:54]: for VS Code, for example, there is this setting you can setup:
image

Comment (furkanmutlu) [reply frontend/package.json:54]: Let me know if that works for you if not I can fix it

**Review by furkanmutlu — COMMENTED** [review]
Fix: Insert new lines

File: frontend/package.json (hunk 1 above)
Comment (ashwanijha1996) [reply frontend/package.json:54]: Hi @furkanmutlu Thanks for your help. I have added new lines at proper places and created a PR to merge those changes. Please review it.

**Comment by furkanmutlu** [comment]
Just push your new commit here

**Comment by furkanmutlu** [comment]
Can you redo all your commits, and create add all your changes into one commit? Then force push it here? 
I think it would be good practice for you to learn as well. Let me know if you can't manage it.

image

**Comment by ashwanijha1996** [comment]
Hi @furkanmutlu i don't know how to redo all my commits. Can i create new PR and Close this PR? If there is another way to do that then it would be great if you will help me out? Thanks.

**Comment by furkanmutlu** [comment]
> Hi @furkanmutlu i don't know how to redo all my commits. Can i create new PR and Close this PR? If there is another way to do that then it would be great if you will help me out? Thanks.

Yes, you can soft reset. 
Since you have 5 commit, RUN:

`git reset --soft HEAD~5`

This will undo 5 last commit from your branch and add them as changes.
Then commit those changes. and do a force push.
`git push -f`

**Pull request closed without merging**